### DISCUSSION TITLE
We're revamping Clockwork!

### DISCUSSION THREAD

**Discussion by ToasterPanic** [red-stone-network/clockwork#33] [Announcements]
We've been working on revamping Clockwork! 

One of the biggest changes for regular users will be to the UI: the entire style has been completely changed! Apps now will all use icons, and instead of removing and closing apps with right-clicks and double-clicks, it'll all now be in a context menu. We'll also be adding app notifications, which allows apps to send notifications directly to Clockwork. You'll also be able to add more than one theme, with a priority system and everything!

For developers, we've made your life a lot easier - the part of apps that are loaded by Clockwork is now written in JSON instead of Javascript. This will allow us to make it easier for developers to create and update their apps, as well as allowing us to keep malicious apps away easier. Don't worry, Javascript apps won't be going anywhere - it'll be turned into the new plugins system.

For more information for developers, you can check the wiki. And if you want access early (developers only!), you can put in a request by emailing toasterpanic@gmail.com.

Thanks for reading!

**Comment by yourmomfsdffddf** [comment]
wait i wanted to ask i am also a begginner in coding and im coding my website and i was wondering if i could ask you how did you make it so when you click the link to your thing it goes into about blank and in a new window and how did you put the nav bar in the bottom

**Reply by ToasterPanic** [reply]
I suggest you check the source for the bookmarklet, try to read through, search up what everything does.

For the about:blank, it just opens a about:blank window, and then sets the innerHTML of the body element to an iframe. 

The navbar uses the `position` and `bottom` CSS properties to move it to the bottom.

For future reference, announcement comments are not the place to ask for help for stuff.

**Reply by yourmomfsdffddf** [reply]
thx and when is the new clock work going to come out

**Comment by Vozo-SigP255** [comment]
When is this expected to be out?

**Reply by ToasterPanic** [reply]
Sometime in May/June.

**Comment by ghost** [comment]
You should add a proxy browser.

**Reply by ToasterPanic** [reply]
You'll actually be able to put all apps through a proxy.

I did also make a Rammerhead Browser app so yeah

**Comment by bro-my-username-got-blocked** [comment]
is there a way to deploy it to a different website?

**Reply by ToasterPanic** [reply]
There will be.
